final_task/rss_reader/functions/convertation.py

```diff
@@ -0,0 +1,204 @@
+    all_links = news.create_list_of_links()
+
+    with doc_html:
+        tag.h1(news.title)
+        with tag.p(news.date):
+            tag.br()
+            tag.a("Link to this article", href=news.link)
+        if internet:
+            for link in images_links:
+                tag.img(src=link)
+        else:
+            for link in images_links:
+                tag.a("Image link", href=link)
+        tag.p(news.text)
+        for num, link in enumerate(all_links):
+            tag.a("Link №{}".format(num + 1), href=link)
+            tag.br()
+    return doc_html
+
+
+def create_epub_object():
+    """Create EpubBook object"""
+    book = epub.EpubBook()
+    book.set_identifier('486464534')
+    book.set_title('News book')
+    book.set_language('en')
+    book.add_author('Anton Zimahorau')
+
+    # define css style
+    style = '''
+    @namespace epub "http://www.idpf.org/2007/ops";
+    body {
+        font-family: Cambria, Liberation Serif, Bitstream Vera Serif, Georgia, Times, Times New Roman, serif;
+    }
+    h2 {
+         text-align: left;
+         text-transform: uppercase;
+         font-weight: 200;     
+    }
+    ol {
+            list-style-type: none;
+    }
+    ol > li:first-child {
+            margin-top: 0.3em;
+    }
+    nav[epub|type~='toc'] > ol > li > ol  {
+        list-style-type:square;
+    }
+    nav[epub|type~='toc'] > ol > li > ol > li {
+            margin-top: 0.3em;
+    }
+    '''
+
+    # add css file
+    nav_css = epub.EpubItem(uid="style_nav", file_name="style/nav.css", media_type="text/css", content=style)
+    book.add_item(nav_css)
+
+    book.spine = ['nav']
+    return book
+
+
+def create_epub(command_line_args, news_collection, logger):
+    """Creates epub book from news collection"""
+    news_collection = limit_news_collections(command_line_args, news_collection, logger)
+
+    book = create_epub_object()
+
+    # Internet connection checking
+    internet = set_internet_flag(logger)
+
+    logger.info("Creating epub News book...")
+    image_number = 0
+    for num, news in enumerate(news_collection):
+        logger.info("Create epub book chapter {}".format(num+1))
+
+        if internet:
+            list_of_images_objects, image_number = create_image_objects(news, image_number)
+            for img_obj in list_of_images_objects:
+                book.add_item(img_obj)
+        else:
+            list_of_images_objects = []
+
+        chapter = epub.EpubHtml(title=news.title, file_name=str(num), media_type="application/xhtml+xml")
+        doc_html = dominate.document(title='RSS News')
+        doc_html = create_html_news_entry_for_epub(news, doc_html, internet, list_of_images_objects)
+        content = str(doc_html)
+
+        chapter.set_content(content)
+        book.add_item(chapter)
+        book.spine.append(chapter)
+
+    book.add_item(epub.EpubNcx())
+    book.add_item(epub.EpubNav())
+
+    path_to_ebook = os.path.join(command_line_args.to_epub, 'News.epub')
```

Comment:
It is better to have a bit more informative filename than this.

The same for html format

Reply:
Fixed